File: base_module/src/main/java/com/ybs/base_module/base/util/Loger.java
package com.ybs.base_module.base.util;

import android.util.Log;

public class Loger {
    private static final boolean DEBUG = true;
    private static final String TAG = "Zzb";
    private static final int THRESHOLD = 2*1024;

    private Loger() {

    }

    private static void handleLog(int level, String msg, String head) {
        StackTraceElement[] stackslist = Thread.currentThread().getStackTrace();
        StackTraceElement tracke = null;

        if (stackslist != null && stackslist.length > 4) {
            tracke = stackslist[4];
        }


        StringBuilder buffer = new StringBuilder(80);

        if (tracke != null) {
            buffer.append('[');
            buffer.append(tracke.getMethodName());
            String fName = tracke.getFileName();
            if (fName == null) {
                buffer.append("(Unknown Source)");
            } else {
                int lineNum = tracke.getLineNumber();

                buffer.append('(');
                buffer.append(fName);
                if (lineNum >= 0) {
                    buffer.append(':');
                    buffer.append(lineNum);
                }
                buffer.append(')');
            }
            buffer.append("]  ");
        }

        if (head != null) {
            buffer.append(head);
        }

        if (msg != null) {
            buffer.append(msg);
        }


        msg = buffer.toString();
        byte[] bytes = msg.getBytes();
        int len =  bytes.length;

        if (len <THRESHOLD) {
            printSubMsg(level, msg);
        } else {

            int count = 0;
            for (int i = 0; i < len; i=(i++)+THRESHOLD) {
                count=Math.min(len-i,THRESHOLD);
                printSubMsg(level,new String(bytes,i,count));
            }

        }




    }

    private static void printSubMsg(int level, String buffer) {
        switch (level) {
            case Log.VERBOSE:
                Log.v(TAG, buffer.toString());
                break;

            case Log.DEBUG:
                Log.d(TAG, buffer.toString());
                break;

            case Log.INFO:
                Log.i(TAG, buffer.toString());
                break;

            case Log.WARN:
                Log.w(TAG, buffer.toString());
                break;

            case Log.ERROR:
                Log.e(TAG, buffer.toString());
                break;

            default:
                break;
        }
    }

    public static void d(String msg) {
        if (DEBUG)
            handleLog(Log.DEBUG, msg, null);
    }

    public static void w(String msg) {
        if (DEBUG)
            handleLog(Log.WARN, msg, null);
    }

    public static void e(String msg) {
        if (DEBUG)
            handleLog(Log.ERROR, msg, null);
    }

    public static void i(String msg) {
        if (DEBUG)
            handleLog(Log.INFO, msg, null);
    }

    public static void v(String msg) {
        if (DEBUG)
            handleLog(Log.VERBOSE, msg, null);
    }

    public static void d(String head, String msg) {
        if (DEBUG)
            handleLog(Log.DEBUG, msg, head);
    }

    public static void i(String head, String msg) {
        if (DEBUG)
            handleLog(Log.INFO, msg, head);
    }

    public static void w(String head, String msg) {
        if (DEBUG)
            handleLog(Log.WARN, msg, head);
    }

    public static void e(String head, String msg) {
        if (DEBUG)
            handleLog(Log.ERROR, msg, head);
    }

    public static void v(String head, String msg) {
        if (DEBUG)
            handleLog(Log.VERBOSE, msg, head);
    }
}

File: base_module/src/main/java/com/ybs/base_module/base/util/Loger.kt
import android.util.Log


/**
 *
 * @ProjectName:    AW
 * @Package:        com.aylson.aw.santos.util
 * @ClassName:      Loger
 * @Description:
 * @Author:         TaxiDriverSantos
 * @CreateDate:     2020/3/12   14:05
 * @UpdateUser:     TaxiDriverSantos
 * @UpdateDate:     2020/3/12   14:05
 * @UpdateRemark:
 * @Version:        1.0
 */

const val head = ""
const val DEBUG = true
const val TAG = "Zzb"
const val THRESHOLD = 2 * 1024  // three byte


fun handleLog(level: Int, msg: String, head: String) {

    val stackTrace = Thread.currentThread().stackTrace
    var trace: StackTraceElement? = null

    if (stackTrace.size > 4) {
        trace = stackTrace[4]
    }

    val sb = StringBuilder(80)

    if (trace != null) {
        sb.append("[ ")
        val fileName = trace.fileName

        if (fileName == null) {
            sb.append("(Unknown Source)")
        } else {
            val lineNumber = trace.lineNumber
            sb.append("(")
            sb.append(fileName)
            if (lineNumber >= 0) {
                sb.append(':')
                sb.append(lineNumber)
            }
            sb.append(")")
            sb.append(" ### ")

            val clzName = trace.className
            val lastName = clzName.split(".").last()
            sb.append(lastName + "." + trace.methodName + "()")
        }
        sb.append(" ] ")
        sb.append("printLog --> ")
    }

    sb.append(head)

    sb.append(msg)


    val log = sb.toString()
    val toByteArray = log.toByteArray()
    val len = toByteArray.size

    if (len < THRESHOLD) {
        printSubMsg(level, log)
    } else {
        var count = 0
        for (i in 0 until len step THRESHOLD) {
            count = (len - i).coerceAtMost(THRESHOLD)
            printSubMsg(level, String(toByteArray, i, count))
        }
    }
}

private fun printSubMsg(level: Int, logStr: String) {
    val lines = logStr.split(System.lineSeparator())
    lines.forEach {
        when (level) {
            Log.VERBOSE -> Log.v(TAG, "--**--**-- START --**--**-- $it")
            Log.DEBUG -> Log.d(TAG, "--**--**-- START --**--**-- $it")
            Log.INFO -> Log.i(TAG, "--**--**-- START --**--**-- $it")
            Log.WARN -> Log.w(TAG, "--**--**-- START --**--**-- $it")
            Log.ERROR -> Log.e(TAG, "--**--**-- START --**--**-- $it")
            else -> Log.e(TAG, "--**--**-- START --**--**-- $it")
        }
    }

}


fun v(msg: String) {
    if (DEBUG) {
        handleLog(Log.VERBOSE, msg, head)
    }
}

fun d(msg: String) {
    if (DEBUG) {
        handleLog(Log.DEBUG, msg, head)
    }
}

fun i(msg: String) {
    if (DEBUG) {
        handleLog(Log.INFO, msg, head)
    }
}

fun w(msg: String) {
    if (DEBUG) {
        handleLog(Log.WARN, msg, head)
    }
}

fun e(msg: String) {
    if (DEBUG) {
        handleLog(Log.ERROR, msg, head)
    }
}